sound/soc/sof/amd/acp.c

```diff
@@ -701,6 +701,10 @@ int amd_sof_acp_probe(struct snd_sof_dev *sdev)
 		goto unregister_dev;
 	}
 
+	ret = acp_init(sdev);
+	if (ret < 0)
+		goto free_smn_dev;
+
```

Comment:
@vijendarmukunda just wondering if you need to call acp_reset() in the error paths below if you make this change

Reply:
@ranj063 : not required. calling acp reset will reset all registers . we want to retain register values during failure.